### PR TITLE
V2 pod top

### PR DESCRIPTION
**Pull request by vrothberg**
The v2 endpoint for pod-top was missing, so we're first adding this endpoint and then do the plumbing for podmanV2.

While CI is cooking, I'll look into ending some tests for the endpoint but I was born after the war and don't speak perl.

### CODE REVIEW

**Comment by mheon** [comment]
LGTM

File: cmd/podmanV2/pods/top.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	topCommand = &cobra.Command{
+		Use:               "top [flags] POD [FORMAT-DESCRIPTORS|ARGS]",
+		Short:             "Display the running processes in a pod",
+		Long:              topDescription,
+		PersistentPreRunE: preRunE,
+		RunE:              top,
+		Args:              cobra.ArbitraryArgs,
+		Example: `podman top podID
+podman top --latest
```
Comment (baude) [inline cmd/podmanV2/pods/top.go]: shoiuld all these be pod top?

File: pkg/api/server/register_pods.go [hunk 2]
```diff
@@ -263,5 +263,36 @@ func (s *APIServer) registerPodsHandlers(r *mux.Router) error {
...
+	//    default: true
+	//    description: Stream the output
+	//  - in: query
+	//    name: ps_args
+	//    type: string
+	//    default: -ef
+	//    description: arguments to pass to ps such as aux. Requires ps(1) to be installed in the container if no ps(1) compatible AIX descriptors are used.
+	// responses:
+	//   200:
+	//     $ref: "#/responses/DockerTopResponse"
```
Comment (baude) [inline pkg/api/server/register_pods.go]: any chance we can NOT have docker in the response description?

File: cmd/podmanV2/pods/top.go (hunk 1 above)
Comment (vrothberg) [reply cmd/podmanV2/pods/top.go]: They should. Good catch!

**Comment by baude** [comment]
couple of non-code things ... else lgtm

File: pkg/api/server/register_pods.go (hunk 2 above)
Comment (vrothberg) [reply pkg/api/server/register_pods.go]: Sure, will change. Inherited from the containers top one, will have a look there as well.

Comment (vrothberg) [reply pkg/api/server/register_pods.go]: Done. PTanotherLook.

**Comment by rhatdan** [comment]
Needs a rebase

**Comment by TomSweeneyRedHat** [comment]
LGTM once rebased.

**Comment by vrothberg** [comment]
Restarted flaked tests.

**Comment by vrothberg** [comment]
Ready to merge

**Pull request merged**